src/libraries/System.Formats.Tar/src/System/Formats/Tar/TarHeader.Read.cs

```diff
@@ -566,15 +566,16 @@ private void ReadUstarAttributes(ReadOnlySpan<byte> buffer)
         // Throws if end of stream is reached or if an attribute is malformed.
         private void ReadExtendedAttributesBlock(Stream archiveStream)
         {
-            if (_size != 0)
+            long size = _size;
+            if (size != 0)
             {
                 ValidateSize();
 
                 byte[]? buffer = null;
-                Span<byte> span = _size <= 256 ?
+                Span<byte> span = (ulong)size <= 256 ?
                     stackalloc byte[256] :
-                    (buffer = ArrayPool<byte>.Shared.Rent((int)_size));
-                span = span.Slice(0, (int)_size);
+                    (buffer = ArrayPool<byte>.Shared.Rent((int)size));
+                span = span.Slice(0, (int)size);
```

Comment:
If the size could actually be greater than Array.MaxValue, is this logic not busted?

Reply:
To be fair, no clue - I presume my changes don't change the behavior here?